pkg/codegen/python/doc.go

```diff
@@ -91,11 +92,28 @@ func elementTypeToName(el string) string {
 // getListWithTypeName returns a Python representation of a list containing
 // items of `t`.
 func getListWithTypeName(t string) string {
-	return fmt.Sprintf("list[%s]", PyName(t))
+	if t == "string" {
+		return "List[str]"
+	}
+
+	return fmt.Sprintf("List[%s]", PyName(t))
 }
 
 // getDictWithTypeName returns the Python representation of a dictionary
 // where each item is of type `t`.
 func getDictWithTypeName(t string) string {
-	return fmt.Sprintf("dict{%s}", PyName(t))
+	return fmt.Sprintf("Dict[%s]", PyName(t))
+}
+
+// getMapWithTypeName returns the Python representation of a dictionary
+// with a key of type `t` and a value of Any.
+func getMapWithTypeName(t string) string {
+	switch t {
+	case "string":
+		return "Dict[str, Any]"
+	case "any":
+		return "Dict[Any, Any]"
+	default:
+		return fmt.Sprintf("Dict[%s, Any]", t)
+	}
```

Comment:
I don't think this is correct. Schema.MapType is defined as:

https://github.com/pulumi/pulumi/blob/ee62f38ab998d7aaaefd16742f8b94912a37c57b/pkg/codegen/schema/schema.go#L101-L105

> MapType represents maps from strings to particular element types. 

So the dictionary's key should always be `str` and the value should be what `t` is specifying.

Reply:
Good catch! Fixed it in the latest commit.